metron-interface/metron-alerts/e2e/login/login.e2e-spec.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { LoginPage } from './login.po';
+
+describe('login to application', function() {
+    let page: LoginPage;
+
+    beforeEach(() => {
+        page = new LoginPage();
+    });
+
+    it('should display error message for invalid credentials', () => {
```

Comment:
The E2E tests are timing out for me.  The first timeout occurs with this new test that was added.   But most of the other tests seem to timeout also.  

The error references this [doc page for Protractor](https://github.com/angular/protractor/blob/master/docs/timeouts.md#waiting-for-angular).  

I tried increasing the timeout from 25 seconds to 60 seconds and it still timed out.  Any ideas?


```
$ npm run e2e

> metron-alerts@0.4.1 pree2e /Users/nallen/tmp/metron-pr699/metron-interface/metron-alerts
> webdriver-manager update

[10:52:03] I/file_manager - creating folder /Users/nallen/tmp/metron-pr699/metron-interface/metron-alerts/node_modules/protractor/node_modules/webdriver-manager/selenium
[10:52:04] I/update - chromedriver: unzipping chromedriver_2.31.zip
[10:52:04] I/update - chromedriver: setting permissions to 0755 for /Users/nallen/tmp/metron-pr699/metron-interface/metron-alerts/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_2.31
[10:52:04] I/update - geckodriver: unzipping geckodriver-v0.18.0.tar.gz
[10:52:04] I/update - geckodriver: setting permissions to 0755 for /Users/nallen/tmp/metron-pr699/metron-interface/metron-alerts/node_modules/protractor/node_modules/webdriver-manager/selenium/geckodriver-v0.18.0

> metron-alerts@0.4.1 e2e /Users/nallen/tmp/metron-pr699/metron-interface/metron-alerts
> protractor

(node:8685) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
[10:52:05] I/launcher - Running 1 instances of WebDriver
[10:52:05] I/direct - Using ChromeDriver directly...
Spec started

  login to application
    ✗ should display error message for invalid credentials
      - Failed: Timed out waiting for asynchronous Angular tasks to finish after 25 seconds. This may be because the current page is not an Angular application. Please see the FAQ for more details: https://github.com/angular/protractor/blob/master/docs/timeouts.md#waiting-for-angular
      While waiting for element with locator - Locator: By(css selector, input.form-control)
    ✓ should login for valid credentials
    ✓ should logout

A Jasmine spec timed out. Resetting the WebDriver Control Flow.
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
  metron-alerts configure table
    ✗ should select columns from table configuration
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.

A Jasmine spec timed out. Resetting the WebDriver Control Flow.
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
  metron-alerts Search
    ✗ should display all the default values for saved searches
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
    ✗ should have all save search controls and they save search should be working
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Failed: No element found using locator: By(css selector, .save-button)
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
    ✗ should populate search items when selected on table
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Failed: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator By(link text, US)
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
    ✗ should delete search items from search box
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Failed: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator By(link text, US)
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
    ✗ should delete first search items from search box having multiple search fields
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Failed: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator By(link text, US)
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
    ✗ manually entering search queries to search box and pressing enter key should search
      - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      - Failed: No element found using locator: By(css selector, .btn-search-clear)

**************************************************
*                    Failures                    *
**************************************************

1) login to application should display error message for invalid credentials
  - Failed: Timed out waiting for asynchronous Angular tasks to finish after 25 seconds. This may be because the current page is not an Angular application. Please see the FAQ for more details: https://github.com/angular/protractor/blob/master/docs/timeouts.md#waiting-for-angular
  While waiting for element with locator - Locator: By(css selector, input.form-control)

2) metron-alerts configure table should select columns from table configuration
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.

3) metron-alerts Search should display all the default values for saved searches
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.

4) metron-alerts Search should have all save search controls and they save search should be working
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Failed: No element found using locator: By(css selector, .save-button)

5) metron-alerts Search should populate search items when selected on table
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Failed: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator By(link text, US)

6) metron-alerts Search should delete search items from search box
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Failed: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator By(link text, US)

7) metron-alerts Search should delete first search items from search box having multiple search fields
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Failed: Index out of bound. Trying to access element at index: 0, but there are only 0 elements that match locator By(link text, US)

8) metron-alerts Search manually entering search queries to search box and pressing enter key should search
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  - Failed: No element found using locator: By(css selector, .btn-search-clear)

Executed 10 of 10 specs (8 FAILED) in 7 mins 13 secs.
[10:59:20] I/launcher - 0 instance(s) of WebDriver still running
[10:59:20] I/launcher - chrome #01 failed 8 test(s)
[10:59:20] I/launcher - overall: 8 failed spec(s)
[10:59:20] E/launcher - Process exited with error code 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! metron-alerts@0.4.1 e2e: `protractor`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the metron-alerts@0.4.1 e2e script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/nallen/.npm/_logs/2017-08-17T14_59_20_570Z-debug.log
```